vignettes/GenomicFiles.Rmd

```diff
@@ -0,0 +1,727 @@
+for available memory.
+
+## Counting read junctions
+
+This example highlights how `reduceByFile` allows detailed control over
+the combination of data subsets from distinct ranges within the same file.
+
+Define ranges of interest.
+
+```{r junctions-ranges}
+gr <- GRanges("chr14", IRanges(c(19100000, 106000000), width=1e7))
+```
+
+The MAP produces a table of junction counts (i.e., 'N' operations in the
+CIGAR) for each range.
+
+```{r junctions-MAP}
+MAP <- function(range, file, ...) {
+     requireNamespace("GenomicAlignments") ## for readGAlignments()
+                                           ## ScanBamParam()
+     param = Rsamtools::ScanBamParam(which=range)
+     gal = GenomicAlignments::readGAlignments(file, param=param)
+     table(GenomicAlignments::njunc(gal))
+}
+```
+
+Create a GenomicFiles object.
+
+```{r junctions-GenomicFiles}
+gf <- GenomicFiles(gr, fls)
+gf
+```
+
+The GenomicFiles object or any subset of the object can be used as the
+ranges argument to functions in `GenomicFiles`. Here the object is
+subset on 3 files and both ranges.
+
+```{r junctions-counts1}
+counts1 <- reduceByFile(gf[,1:3], MAP=MAP)
+length(counts1) ## 3 files
+elementNROWS(counts1) ## 2 ranges
+```
+
+Each list element has a table of counts for each range.
+
+```{r junctions-counts1-show}
+counts1[[1]]
+```
+
+Add a reducer that combines counts for records in each range with
+exactly 1 junction.
+
+```{r junctions-REDUCE}
+REDUCE <- function(mapped, ...)
+    sum(sapply(mapped, "[", "1"))
+    reduceByFile(gr, fls, MAP, REDUCE)
+```
+
+Next invoke `reduceFiles` with the same files and MAP function.
+`reduceFiles` treats all ranges as a group and counts junctions for all
+ranges simultaneously.
+
+```{r junctions-counts2}
+counts2 <- reduceFiles(gf[,1:3], MAP=MAP)
+```
+
+In the `reduceByFile` example junctions were counted for each range
+individually which allowed us to see results for the individual ranges
+and combine them on the fly based on specific criteria. In contrast,
+`reduceFiles` counts junctions for all ranges simultaneously.
+
+```{r junctions-counts2-show}
+## reduceFiles returns counts for all ranges.
+counts2[[1]]
+## reduceByFile returns counts for each range separately.
+counts1[[1]]
+```
+
+## Coverage 1: reduceByFile
+
+Files that are too large to fit in memory can be streamed over by creating
+'tiles' or ranges that span the whole file. The `tileGenome` function creates a
+set of continuous ranges that span a given seqlength(s). The sample BAM files
+contain only chr14 so we extract the appropriate seqlength from the BAM files
+and use it in`tileGenome`. In this example we create 5 ranges but the optimal
+value for `ntile` will depend on the application and the size of the chromosome
+(or genome) to be tiled.
+
+```{r coverage1-tiles}
+chr14_seqlen <- seqlengths(seqinfo(BamFileList(fls))["chr14"])
+tiles <- tileGenome(chr14_seqlen, ntile=5)
+```
+
+tiles is a `GRangesList` of length `ntile` with one range per element
```

Comment:
Tiles should have backticks. In the PDF, GRangesList doesn't have special formatting but I think it should so let's leave GRangesList as it is in your .Rmd.